app/assets/javascripts/votes.js

```diff
@@ -0,0 +1,115 @@
+$(document).ready(function() {
+  var flag = true;
+  var flag2 = true;
+  $(".upvote").on("click", function( event ) {
+    if (flag) {
+      event.preventDefault();
+      event.stopPropagation();
+
+      var reviewId = event.toElement.className.match(/\d+/)[0];
+
+      var score = document.getElementById("review-" + reviewId + "-score").innerHTML;
+      score = parseInt(score);
+      score++;
+      document.getElementById("review-" + reviewId + "-score").textContent = score;
+
+      // also hide upvote button
+      var url = '/reviews/' + reviewId + '/upvotes';
+
+      $.ajax({
+        type: "POST",
+        url: url,
+        dataType: "json",
+        complete: function(upvote) {
+          var upvoteId = JSON.parse(upvote.responseText)["id"];
+          var reviewClass = ".review-" + reviewId + ".upvote";
+          $(reviewClass).attr("data-method", "delete");
+          $(reviewClass).text("Delete Upvote");
+        }
+      });
+    }else{
+      event.preventDefault();
+      event.stopPropagation();
+
+      var reviewId = event.toElement.className.match(/\d+/)[0];
+
+      var score = document.getElementById("review-" + reviewId + "-score").innerHTML;
+      score = parseInt(score);
+      score--;
+      document.getElementById("review-" + reviewId + "-score").textContent = score;
+
+      var reviewClass = ".review-" + reviewId + ".upvote";
+      var upvoteId = $(reviewClass).attr("href").match(/\d+/)[0];
+
+      // also hide upvote button
+      var url = '/reviews/' + reviewId + '/upvotes/' + upvoteId;
+
+      $.ajax({
+        type: "DELETE",
+        url: url,
+        dataType: "json",
+        complete: function(upvote) {
+          $(reviewClass).attr("data-method", "post");
+          $(reviewClass).text("Upvote");
+        }
+      });
+    }
+    flag = !flag;
+  });
+
+  $(".downvote").on("click", function( event ) {
+    if (flag2) {
+      event.preventDefault();
+      event.stopPropagation();
+
+      var reviewId = event.toElement.className.match(/\d+/)[0];
+
+      var score = document.getElementById("review-" + reviewId + "-score").innerHTML;
+      score = parseInt(score);
+      score--;
+      document.getElementById("review-" + reviewId + "-score").textContent = score;
+
+      // also hide downvote button
+      var url = '/reviews/' + reviewId + '/downvotes';
+
+      $.ajax({
+        type: "POST",
+        url: url,
+        dataType: "json",
+        complete: function(downvote) {
+          var downvoteId = JSON.parse(downvote.responseText)["id"];
```

Comment:
['id'] is better written in dot notation.<br>'downvoteId' is defined but never used.